config/gateway.go

```diff
@@ -53,6 +53,12 @@ type Gateway struct {
 	//  }
 	PathPrefixes []string
 
+	// MaxDirectorySize is the maximum directory size that will be listed in the
+	// gateway. A directory being listed with a detected size above this threshold
+	// will be stopped to avoid performance downgrade while fetching entries'
+	// metadata. The default value of 0 means no limit will be applied.
+	MaxDirectorySize int `json:",omitempty"`
```

Comment:
Let's make this optional in the config, most users will be fine with implicit default.

```suggestion
	MaxDirectorySize *OptionalInteger `json:",omitempty"`
```

Reply:
Nice, didn't know this existed.

Reply:
I'm not sure `OptionalInteger` added anything different than ``json:",omitempty"`` if we're using the default Go value of 0. @lidel Could you confirm how did you envision us using this? I'm basically calling it like `MaxDirectorySize: int(cfg.Gateway.MaxDirectorySize.WithDefault(0)),`.

Reply:
We will be using it with value other than  0, something like `.WithDefault(2500)`. 
That way we can adjust it in future releases without the need for writing config migration.